packages/database/src/core/view/filter/LimitedFilter.ts

```diff
@@ -300,4 +292,26 @@ export class LimitedFilter implements NodeFilter {
       return snap;
     }
   }
+
+  private withinDirectionalStart = (node: NamedNode) =>
+    this.reverse_ ? this.withinEndPost(node) : this.withinStartPost(node);
+
+  private withinDirectionalEnd = (node: NamedNode) =>
+    this.reverse_ ? this.withinStartPost(node) : this.withinEndPost(node);
+
+  private withinStartPost = (node: NamedNode) => {
+    const compareRes = this.index_.getCompare()(
```

Comment:
nit: I don't like having two parentheses side by side, could we just use compare directly here?

Reply:
Done.